javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentClassLoader.java

```diff
@@ -61,19 +62,19 @@ public AgentClassLoader(
       addURL(new URL("x-internal-jar", null, 0, "/", internalJarUrlHandler));
     } catch (MalformedURLException e) {
       // This can't happen with current URL constructor
-      log.error("URL malformed.  Unsupported JDK?", e);
+      throw new IllegalStateException("URL malformed.  Unsupported JDK?", e);
     }
 
     if (!AGENT_INITIALIZER_JAR.isEmpty()) {
       URL url;
       try {
         url = new File(AGENT_INITIALIZER_JAR).toURI().toURL();
       } catch (MalformedURLException e) {
-        log.warn(
+        throw new IllegalStateException(
```

Comment:
This only happens due to a bug in packaging and we assume that a any published distribution at least has enough testing to know it won't cause this crash I guess. It seems fine to me.